client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -359,6 +359,8 @@ public void close() {
             }
         }
         ExecutorServiceHelpers.shutdown(retransmitPool);
+        controller.removeWriter(writerId, stream)
+                .thenAccept(r -> log.info("Writer {} has been marked idle", writerId));
```

Comment:
Marking the writer as idle during close method would help in excluding this writer from timestamp aggregation calculation. This way it would not wait for this writer to report its's marks until TimestampAggregationTimeout occurs.